tsdb/raw.go

```diff
@@ -93,6 +93,7 @@ func (e *RawExecutor) execute(out chan *models.Row) {
 	// Keep looping until all mappers drained.
 	var err error
 	for {
+		timeout := time.After(30 * time.Second)
```

Comment:
If you move the timeout start time up here then you could potentially have long running mappers that time it out.